lib/Sema/TypeCheckCompilerEvaluable.cpp

```diff
@@ -26,26 +26,205 @@ using namespace swift;
 
 namespace {
 
-/// Checks that the body of a function is compiler evaluable.
-/// Currently a skeleton implementation that only rejects while loops.
+/// Checks that a type is compiler representable.
```

Comment:
what does "compiler representable" mean -- is it defined/documented somewhere?

Reply:
the proposal describes it in this section: https://gist.github.com/marcrasi/b0da27a45bb9925b3387b916e2797789#proposed-solution---intuitive-approach

I also plan to add more detailed comments describing "compiler representable" in a PR where I implement a real check for it.

Reply:
Thanks. Why is float not compiler representable?

Reply:
Float's not representable because we didn't propose to support it in the proposal. I think there are no fundamental difficulties with it -- we can just add the support whenever we want.